experimental/javareach/reachable.go

```diff
@@ -59,12 +59,21 @@ func NewReachabilityEnumerator(
 	}
 }
 
-func (r *ReachabilityEnumerator) EnumerateReachabilityFromClasses(mainClasses []string) (*ReachabilityResult, error) {
+func (r *ReachabilityEnumerator) EnumerateReachabilityFromClasses(mainClasses []string, serviceClasses []string) (*ReachabilityResult, error) {
```

Comment:
actually I'd recommend renaming `serviceClasses` to `optionalRootClasses` or something similar to decouple the logic here a bit.